src/utilities/util_dump.c

```diff
@@ -652,18 +661,60 @@ dump_record(WT_CURSOR *cursor, bool reverse, bool json)
         infix = "\n";
         suffix = "\n";
     }
-    while ((ret = (reverse ? cursor->prev(cursor) : cursor->next(cursor))) == 0) {
-        if ((ret = cursor->get_key(cursor, &key)) != 0)
-            return (util_cerr(cursor, "get_key", ret));
-        if ((ret = cursor->get_value(cursor, &value)) != 0)
-            return (util_cerr(cursor, "get_value", ret));
-        if (fprintf(
-              fp, "%s%s%s%s%s%s", json && once ? "," : "", prefix, key, infix, value, suffix) < 0)
-            return (util_err(session, EIO, NULL));
-        once = true;
+
+    while (ret == 0) {
+
+        /* A specific key is requested. */
+        if (key != NULL) {
+            current_key = key;
+            cursor->set_key(cursor, current_key);
+
+            if (search_near)
+                ret = cursor->search_near(cursor, &exact);
+            else
+                ret = cursor->search(cursor);
+
+            if (ret != 0 && ret != WT_NOTFOUND)
```

Comment:
I'm curious what WT_NOTFOUND means for search_near vs non-search_near cases. Do we need special handling for the `ret == WT_NOTFOUND` condition here?

Reply:
I have refactored the code here bc2541da366416490db7c8d18cabb9a3f6dd998c